### PR TITLE
feat: jackson provider

### PR DESCRIPTION
**Pull request by hperl**
This adds a jackson provider to Kratos.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: embedx/config.schema.json [hunk 1]
```diff
@@ -670,6 +670,49 @@
...
     },
+    "selfServiceSAMLProvider": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "examples": ["be2b30", "my-saml-provider"]
+        },
+        "label": {
+          "title": "Optional string which will be used when generating labels for UI buttons.",
```
Comment (aeneasr) [inline embedx/config.schema.json:681]:
```suggestion
          "title": "Label",
          "description": "Optional string which will be used when generating labels for UI buttons.",
```

File: embedx/config.schema.json [hunk 2]
```diff
@@ -1962,6 +2015,39 @@
                   }
                 }
               }
+            },
+            "saml": {
```
Comment (aeneasr) [inline embedx/config.schema.json:2019]: Add a description when this works

File: selfservice/strategy/oidc/strategy_registration.go [hunk 3]
```diff
@@ -164,6 +164,13 @@ func (s *Strategy) Register(w http.ResponseWriter, r *http.Request, f *registrat
 		span.SetAttributes(attribute.String("not_responsible_reason", "provider ID missing"))
 		return errors.WithStack(flow.ErrStrategyNotResponsible)
 	}
+	// This is a hack for a lack of a `method` field in the form body.
+	if prefix, _, ok := strings.Cut(pid, ":"); ok {
```
[the remark this answers is not in the document]
Comment (aeneasr) [reply selfservice/strategy/oidc/strategy_registration.go:168]: Why is the code still here?

**Review by aeneasr — COMMENTED** [review]
Two comments otherwise LGTM

**Review by aeneasr — APPROVED** [review]

**Comment by 3kwtloo** [comment]
@hperl thank you for adding this!

I am new to kratos and currently building a PoC where I am connecting to jackson with version 1.3.1 through the generic OIDC provider as of now and I am looking forward to use this provider in the future.  

I might be completely wrong but looking at this implementation seems to assume that Jackson is only handling SAML connections but it also supports connecting to OIDC through it as well.



As a side note: I was also wondering if there is a nice way to pass some form of login_hint with tenantId and productId to this provider dynamically so that it could be used in a more generic way since jackson can use tenant=<tenantID>&product=<productID> as a client_id.
This would allow to dynamically add B2B connections without having to change the kratos config and restarting the server.

**Pull request merged**